lacommunaute/forum_upvote/views.py

```diff
@@ -17,48 +18,72 @@
 track_handler = TrackingHandler()
 
 
-class PostUpvoteView(PermissionRequiredMixin, View):
+class BaseUpvoteMixin:
     def dispatch(self, request, *args, **kwargs):
         if request.method != "POST":
             return self.http_method_not_allowed(request)
         return super().dispatch(request, *args, **kwargs)
 
-    def get_object(self):
-        if not hasattr(self, "object"):
-            self.object = get_object_or_404(
-                Post.objects.select_related("topic__forum").all(),
-                pk=self.request.POST["post_pk"],
-                topic__pk=self.request.POST["pk"],
-            )
-        return self.object
+    def perform_permissions_check(self, user, obj, perms):
+        return self.request.forum_permission_handler.can_read_forum(obj, user)
 
     def post(self, request, **kwargs):
-        post = self.get_object()
+        obj = self.get_object()
```

Comment:
Astuce pour éviter de répéter les requêtes, on peut stocker l’objet sur la vue parmi les premières instructions. Ça évite d’avoir à implémenter un cache, est on peut être sûrs que self.object est défini.
https://github.com/django/django/blob/86561844ce66cda3e6a8c22d4ace4c2d1bc1f2e7/django/views/generic/detail.py#L108
```suggestion
        self.object = self.get_object()
```